### PR TITLE
set up a dialog box to help with import/export

### LINKED ISSUE #475: better import/export
Once we merge feature/xls.import and feature/xls.export, here's what I'd like to enable on top of the current functionality:

**Released in v2.8**

- [x] enable xls/csv export
- [x] enable copy-paste import
- [x] firm up all map API endpoints
- [x] consider permissions during import/export
- [x] make x and y optional during import
- [x] metacodes SHOW api endpoint
- [x] improve the serialization format for csv/xls
- [x] implement json export
- [x] re-implement import with the new format
- [x] add a json parser to the JS import module

**(will be) Released in 2.10**

- [x] drag and drop files triggers import
- [x] add a csv parser to the JS import module
- [x] icon on top right for import with
  - [x] instructions on copy-paste
  - [x] instructions + button to export map
  - [x] file upload field for csv files
- [ ] browser testing

** Decided to drop **

- [x] ~~upgrade tsv parser in the JS import module~~
- [x] ~~file upload parser in the JS import module for xls/csv/ods/xlsx?~~

### PR DESCRIPTION
**Pull request by devvmh**
ready to merge, but i would like a better icon

- [x] Use a different icon than the star icon
- [x] redesign the look and feel of the box to be less ugly

Merging this PR closes #475.

### CODE REVIEW

**Comment by Bortseb** [comment]
I have a few thoughts and questions on this issue after chatting with @devvmh on #671 (Read my last comment there first if you haven't already)

Is the idea that the dialog will pop up if someone pastes or drags content into the map? I think this would be useful because we could then say something like "Hey, we think you are pasting CSV-type data, and that would result in 25 topics being created. Is this your intention?"

Also, if the user is dragging a lot of links into the map from another tab... we could add a checkbox that says something like "Stop confirming my imported data types with this dialog for the rest of this session" (or for [input number] more imports)

**Comment by devvmh** [comment]
Ooh, that's a cool idea. Currently this dialog pops up after clicking a little button next to Star, Map Info, and Help in the bottom right.

What if import happened automatically in most cases, and in ambiguous cases it popped up?

I'm happy to postpone merging this until we have a great design for it. @poietic has done a design in the past I think too

**Comment by Bortseb** [comment]
I'm really excited to start using import, so would be for it to continually improved instead of making quantum leaps.

**Comment by devvmh** [comment]
I kind of like import where it is now, though, which is available but undocumented. I'd love for you to start using it as-is on heroku. It is pretty functional, honestly, and all we'd really be adding is UI

Once we do add a UI, though, then we need to keep supporting whatever functionality we have enabled at that time. So I'm in favour of holding off on the UI until it's really great and user friendly, but leaving it as an open secret and telling interested people about it early. Does that make sense?

In other words, having you use Import before we build a UI is really valuable, so we can use the insights you get to build a better interface.

**Comment by Bortseb** [comment]
oh! see, I don't know the secret... I tried hitting paste, but nothing seemed to happen... Whats the trick?

**Comment by devvmh** [comment]
if you copy-paste from Excel right now and paste into metamaps.cc it should work. If you copy-paste JSON from the map export view, that should work too

if you drag files on to metamaps.herokuapp.com it should work, whether it's tsv, csv, or json. We can troubleshoot on slack

File: app/assets/stylesheets/application.scss.erb [hunk 1]
```diff
@@ -2028,17 +2055,17 @@ and it won't be important on password protected instances */
     left: 0;
     width: 100%;
     height: 100%;
-    position: fixed;
+    position: absolute;
```
Comment (devvmh) [inline app/assets/stylesheets/application.scss.erb]: This is a big change, one of a few which allows scrolling in lightboxes. I haven't tested this very thoroughly on other lightboxes

File: app/assets/stylesheets/clean.css.erb [hunk 2]
```diff
@@ -188,7 +188,7 @@
 .upperRightIcon {
     width: 32px;
     height: 32px;
-    background-image: url(<%= asset_data_uri('topright_sprite.png') %>);
+    background-image: url(<%= asset_path('topright_sprite.png') %>);
```
Comment (devvmh) [inline app/assets/stylesheets/clean.css.erb]: This is just an update unrelated to the PR; I got happy using asset_data_uri during the rails 4 upgrade, but I think it's better to use asset_path

**Comment by devvmh** [comment]
I'm merging this so someone will see the icon and make it better hahaha
